cocos2d/actions/CCAction.js

```diff
@@ -528,19 +528,22 @@ cc.Follow = cc.Action.extend({
     },
 
     step:function (dt) {
-        var tempPosX = this._followedNode.x;
-        var tempPosY = this._followedNode.y;
-        tempPosX = this._halfScreenSize.x - tempPosX;
-        tempPosY = this._halfScreenSize.y - tempPosY;
+        var targetWorldPos = this.target.convertToWorldSpaceAR(cc.Vec2.ZERO);
+        var followedWorldPos = this._followedNode.convertToWorldSpaceAR(cc.Vec2.ZERO);
+        // Calculation of Action and follow the offset value
+        var delta = cc.pSub(targetWorldPos, followedWorldPos);
+        // Calculation Centre Point and follow the offset value
+        var offset = cc.pSub(this._halfScreenSize, followedWorldPos);
```

Comment:
followedWorldPos + delta + offset
= followedWorldPos + delta + this._halfScreenSize - followedWorldPos
= delta + this._halfScreenSize